fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraBinaryImpl.java

```diff
@@ -445,6 +448,44 @@ private static void decorateContentNode(final Node contentNode, final Collection
     }
 
     @Override
+    public FedoraResource getTimeMap() {
+        try {
+            final Node container = getContainingNode(getNode()).get();
+            return Optional.of(container.getNode(LDPCV_TIME_MAP + "/" + getNode().getParent().getName()))
+                .map(nodeConverter::convert).orElse(null);
+        } catch (PathNotFoundException e) {
+            throw new PathNotFoundRuntimeException(e);
+        } catch (RepositoryException e) {
+            throw new RepositoryRuntimeException(e);
+        }
+    }
+
+    @Override
+    public FedoraResource findOrCreateTimeMap() {
+        final Node ldpcvParentNode;
+        final Node ldpcvNode;
+
+        try {
+            final Node container = getContainingNode(getNode()).get();
+            ldpcvParentNode = findOrCreateChild(container, LDPCV_TIME_MAP, NT_FOLDER);
```

Comment:
As a comment to this method, could you please detail the design of the TimeMap containers?

Reply:
As we discussed during the sprint wrap-up meeting this morning, the TimeMap for binary will be created inside the TimeMap of the parent container that holds the binary resource, which has the same name as the binary node.